astropy/io/fits/hdu/compressed/tests/test_tiled_compression.py

```diff
@@ -65,6 +66,95 @@ def test_zblank_support(canonical_data_base_path, tmp_path):
         assert_equal(np.round(hdul[1].data), reference)
 
 
+def test_integer_blank_support(canonical_data_base_path, tmp_path):
+    # This uses a test 64x64 section of the m13 image to which three NaN values are added.
+    # It is converted to an integer image with blank values and compressed.
+    # Then 2 variations of the BLANK/ZBLANK specification are created by
+    # editing the header and compressed data table.
+    # When these are read in, the result is a float image that should be the same
+    # as the original (including NaNs for blanks).
+
+    with fits.open(canonical_data_base_path / "m13.fits") as hdul:
+        # pick a 64x64 pixel subset
+        data = hdul[0].data[116:180, 116:180]
+        header = hdul[0].header.copy()
+        del header["CHECKSUM"]
+        del header["DATASUM"]
+
+    # float version of image with NaNs as blanks
+    reference = data.astype(np.float32)
+    reference[[1, 2, 3], [1, 2, 3]] = np.nan
+
+    # set blanks in the int16 image and BLANK keyword in header
+    blank = -32768
+    data[np.isnan(reference)] = blank
+    header["BLANK"] = blank
+
+    # create the compressed file
+    cfile1 = tmp_path / "compressed_with_blank.fits"
+    w = np.where(np.isnan(data))
+    blank = -32768
+    data[w] = blank
```

Comment:
Is not `data` still an integer array at this point? So `np.isnan(data)` should be empty and no `blank`s added to those already set in L90, or am I missing something here?

Reply:
Absolutely correct, as you mentioned above those lines are completely redundant and don't do anything!  I deleted them in the just-submitted version.